src/Yasumi/Provider/Brazil.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ *  This file is part of the Yasumi package.
+ *
+ *  Copyright (c) 2015 - 2016 AzuyaLabs
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ *
+ *  @author Dorian Neto <doriansampaioneto@gmail.com>
+ */
+namespace Yasumi\Provider;
+
+use DateInterval;
+use DateTime;
+use DateTimeZone;
+use Yasumi\Holiday;
+
+/**
+ * Provider for all holidays in the Brazil.
+ */
+class Brazil extends AbstractProvider
+{
+
+    use CommonHolidays, ChristianHolidays;
+
+    /**
+     * Initialize holidays for the Brazil.
+     */
+    public function initialize()
+    {
+        $this->timezone = 'America/Fortaleza';
+
+        // Add common holidays
+        $this->addHoliday($this->newYearsDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->internationalWorkersDay($this->year, $this->timezone, $this->locale));
+
+        // Add Christian holidays
+        $this->addHoliday($this->christmasDay($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->easter($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->corpusChristi($this->year, $this->timezone, $this->locale));
+        $this->addHoliday($this->goodFriday($this->year, $this->timezone, $this->locale));
+
+        /**
+         * Carnaval
+         *
+         * Carnaval do Brasil é a maior festa popular do país. A festa acontece durante quatro dias. (que precedem a
+         * quarta–feira de cinzas). O último dia de Carnaval precede a quarta-feira de cinzas (início da Quaresma).
+         *
+         * @link https://pt.wikipedia.org/wiki/Carnaval_do_Brasil
+         */
+        if ($this->year >= 1700) {
+            $this->addHoliday(new Holiday('carnaval', [],
+                $this->calculateEaster($this->year, $this->timezone)->sub(new DateInterval('P47D')), $this->locale));
+        }
```

Comment:
I believe Carnaval is an observed holiday, not an official/national (by law) holiday. If so, please add the appropriate parameter `HOLIDAY::TYPE_OBSERVANCE` to the **addHoliday** method.